Add docs for connecting to MetaMask (#1525)
MetaMask/metamask-docs: wallet/how-to/use-non-evm-networks/starknet-snap/connect-to-starknet.md

---
description: Connect your Starknet account to MetaMask
sidebar_position: 1
---

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

# Connect your Starknet account to MetaMask

> To connect a dapp to MetaMask for StarkNet, you can use `get-starknet`. 

Reviewer: We should make it clear who the audience is. When reading I assume the audience is 

> A starknet dApp developer that wants to have compatibility with Metamask. 

Allowing metamask users to access the dApp by simply visiting its website with metamask installed and installing the starknet-snap when requested. 

For example AVNU : https://www.avnu.fi. Is compatible with Metamask thanks to the starknet-snap. Users that go to https://app.avnu.fi/ and click connect can choose metamask and will be prompted to install the starknet-snap. 

Behind the scene AVNU uses get-starknet.

Reviewer: If the audience is a user, then we can mention get-starknet because it is interesting in term of interoperability but we should make it clear that a user will never have to directly interact with get-starknet. A user simply installs the snaps when requested.

> This tool simplifies the process of integrating Starknet with MetaMask and allows you to connect your wallets and install the necessary Snap.

Reviewer: If the audience is the developer this needs to be rephrased. 

```suggestion
This tool simplifies the process of integrating Starknet with MetaMask and allows users to connect to starknet dApp with their Metamask wallets.
```

> After set up, the Snap enables secure and efficient interaction with the Starknet ecosystem.

Reviewer 2: We should add the `wallet_invokeSnap` instructions to this page instead of the Starknet landing page. So include a brief blurb here in the intro explaining the use case for each method.

## Prerequisites

- [MetaMask installed](https://metamask.io/download/)
- A text editor (for example, [VS Code](https://code.visualstudio.com/))
- [Node](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) version 20.11 or later
- [Yarn](https://yarnpkg.com/)
- A JavaScript or TypeScript project set up 

## Add `get-starknet` to your project

Add both the [`get-starknet`](https://github.com/MetaMask/snaps/tree/main/packages/create-snap) library and the latest version of the `starknet.js`  library to your project's dependencies:

<Tabs>
  <TabItem value="yarn" label="Yarn" default>

  ```bash
  yarn add get-starknet starknet@next
  ```

  </TabItem>

  <TabItem value="npm" label="npm">

  ```bash
  npm install get-starknet starknet@next
  ```

  </TabItem> 

</Tabs>

## Start the Snap

From the root of the created project, install the project dependencies:

<Tabs>
  <TabItem value="yarn" label="Yarn" default>

  ```bash
  yarn start
  ```

  </TabItem>

  <TabItem value="npm" label="npm">

  ```bash
  npm start
  ```

  </TabItem> 

</Tabs>

> ## Connect to the Snap
> 
> On the  dapp, click **Connect** and the MetaMask extension is displayed and prompts you to approve the Snap's permissions.
> 
> Once connected, select **Send message** to display a custom message within a confirmation
> dialog in MetaMask Flask.

Reviewer 2: ```suggestion
### 3. Connect to the Snap

On the dapp, select **Connect**. The MetaMask extension displays and prompts you to approve the Snap's permissions.

Once connected, select **Send message** to display a custom message within a confirmation
dialog in MetaMask.
```

I don't think this guide requires MetaMask Flask, does it?
